doc/en/deprecations.rst

```diff
@@ -19,6 +19,24 @@ Below is a complete list of all pytest features which are considered deprecated.
 :class:`_pytest.warning_types.PytestWarning` or subclasses, which can be filtered using
 :ref:`standard warning filters <warnings>`.
 
+``junit_family`` default value change to "xunit2"
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. deprecated:: 5.2
+
+The default value of ``junit_family`` option will change to ``xunit2`` in pytest 6.0, given
+that this is the version supported by default in modern tools that manipulate this type of file.
+
+In order to smooth the transition, pytest will issue a warning in case the ``--junitxml`` option
+is given in the command line but ``junit_family`` is not explicitly configured in ``pytest.ini``.
+
+In order to silence this warning, users just need to configure the ``junit_family`` option:
+
+.. code-block:: ini
+
+    [pytest]
+    junit_family=legacy
```

Comment:
the current default is `xunit1` -- should that be listed here (and in the warning message)?

Reply:
xunit1 and legacy are aliases here, I decided to suggest people to use "legacy" so it would convey better that is something older, but I'm fine either way. Let me know if you still prefer xunit1.

Good idea about showing the warning here though. 👍

Reply:
sounds good!  I admittedly didn't know that they were aliases 🙃

Reply:
OK!

Added the warning message. 👍